guest-agent/src/guest_api_service.rs

```diff
@@ -112,6 +115,53 @@ impl GuestApiRpc for GuestApiHandler {
     async fn list_containers(self) -> Result<ListContainersResponse> {
         list_containers().await
     }
+
+    async fn pre_backup(self) -> Result<()> {
+        fs::OpenOptions::new()
+            .create_new(true)
+            .write(true)
+            .open(BACKUP_LOCK_FILE)
+            .context("Failed to create backup lock file, there is another backup in progress")?;
+        // Run /dstack/hooks/pre-backup if it exists
+        let pre_backup_hook = "/dstack/hooks/pre-backup";
```

Comment:
How can we set up hooks for pre-backup and post-backup?

Reply:
Are they set up only by node operators? I'm wondering who can actually create these hook scripts.

Reply:
> Are they set up only by node operators? I'm wondering who can actually create these hook scripts.

It's up to app defines the logic in the hooks. For example, flush their mysql database in the CVM. The hooks is put their for future use. I think we don't need to care about it at this time, until their is a use case appears.